File: Day275_2025/Q275_Leetcode.java
class Q275_Leetcode {
    public int maxBottlesDrunk(int numBottles, int numExchange) {
        int ans = numBottles;
        int empty = numBottles;

        while (empty >= numExchange) {
            ans++;
            empty -= numExchange - 1; // spend numExchange, but gain 1 bottle
            numExchange++;            // exchange cost increases
        }

        return ans;
    }
}

File: Day275_2025/Q275_Leetcode.cpp
class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        // Q. Water Bottles II
        int ans = numBottles;
        for (int empty = numBottles; empty >= numExchange; numExchange++) {
            ans++;
            empty -= numExchange - 1;
        }
        return ans;
    }
};